packages/dd-trace/src/opentelemetry/metrics/periodic_metric_reader.js

```diff
@@ -99,7 +109,27 @@ class PeriodicMetricReader {
     const allMeasurements = this.#measurements.splice(0)
 
     for (const instrument of this.#observableInstruments) {
-      allMeasurements.push(...instrument.collect())
+      if (allMeasurements.length >= this.#maxQueueSize) break
```

Comment:
This could only be true in case the allMeasurements is already exceeding the queue size before the instrument iteration starts. I would move it outside therefore.

And can that case be true at all?

Reply:
I cleaned this up